Mage.Sets/src/mage/sets/fallenempires/DelifsCube.java

```diff
@@ -0,0 +1,119 @@
+import mage.abilities.common.AttacksAndIsNotBlockedTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.RemoveCountersSourceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.RegenerateTargetEffect;
+import mage.abilities.effects.common.continuous.AssignNoCombatDamageSourceEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.cards.CardImpl;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.Rarity;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ *
+ * @author MarcoMarin
+ */
+public class DelifsCube extends CardImpl {
+
+    public DelifsCube(UUID ownerId) {
+        super(ownerId, 170, "Delif's Cube", Rarity.RARE, new CardType[]{CardType.ARTIFACT}, "{1}");
+        this.expansionSetCode = "FEM";
+
+        Ability ability2 = new AttacksAndIsNotBlockedTriggeredAbility(new DelifsCubeEffect(this.getId()), false);        
+        ability2.addEffect(new AssignNoCombatDamageSourceEffect(Duration.EndOfTurn));        
+        // {2}, {tap}: This turn, when target creature you control attacks and isn't blocked, it assigns no combat damage this turn and you put a cube counter on Delif's Cube.
+        SimpleActivatedAbility ability = new SimpleActivatedAbility(Zone.BATTLEFIELD,
+                new GainAbilityTargetEffect(ability2, Duration.EndOfTurn),
+                new ManaCostsImpl("{2}"));
+        ability.addCost(new TapSourceCost());
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        
+        this.addAbility(ability);
+        // {2}, Remove a cube counter from Delif's Cube: Regenerate target creature.
+        SimpleActivatedAbility ability3 = new SimpleActivatedAbility(Zone.BATTLEFIELD,
+                new RegenerateTargetEffect(),
+                new ManaCostsImpl("{2}"));
+        ability3.addCost(new RemoveCountersSourceCost(CounterType.CUBE.createInstance()));
+        ability3.addTarget(new TargetControlledCreaturePermanent());
+        
+        this.addAbility(ability3);
+    }
+
+    public DelifsCube(final DelifsCube card) {
+        super(card);
+    }
+
+    @Override
+    public DelifsCube copy() {
+        return new DelifsCube(this);
+    }
+}
+
+class DelifsCubeEffect extends OneShotEffect{
+                      
+    private UUID cubeId;
+    
+    public DelifsCubeEffect(UUID cubeId) {
+        super(Outcome.Benefit);
+        this.cubeId = cubeId;
+        this.setText("This turn, when target creature you control attacks and isn't blocked, it assigns no combat damage this turn and you put a cube counter on Delif's Cube.");
+    }
```

Comment:
- you can use staticText = "...
- always use {this} instead of the name of the card
- effect text have no need for a dot at the end. They are added automatically if the rule text of all effects of an ability are concatenated, so it's better (and more uniform) to exclude the dots at the end of the text.